tests/test_packages/test_connections/test_ledger_api/test_contract_api.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests of the ledger API connection for the contract APIs."""
+import asyncio
+import json
+from pathlib import Path
+from typing import cast
+
+import pytest
+
+from aea.components.loader import load_component_from_config
+from aea.configurations.base import (
+    ComponentConfiguration,
+    ComponentType,
+    ContractConfig,
+)
+from aea.connections.base import Connection
+from aea.contracts import contract_registry
+from aea.crypto.fetchai import FetchAICrypto
+from aea.crypto.wallet import CryptoStore
+from aea.identity.base import Identity
+from aea.mail.base import Envelope
+
+from packages.fetchai.connections.ledger_api.contract_dispatcher import (
+    ContractApiDialogues,
+)
+from packages.fetchai.protocols.contract_api import ContractApiMessage
+
+from ....conftest import ETHEREUM_ADDRESS_ONE, ROOT_DIR
+
+
+@pytest.fixture()
+async def ledger_apis_connection(request):
+    identity = Identity("name", FetchAICrypto().address)
+    crypto_store = CryptoStore()
+    directory = Path(ROOT_DIR, "packages", "fetchai", "connections", "ledger_api")
+    connection = Connection.from_dir(
+        directory, identity=identity, crypto_store=crypto_store
+    )
+    connection = cast(Connection, connection)
+    await connection.connect()
+    yield connection
+    await connection.disconnect()
+
+
+@pytest.fixture()
+def load_erc1155_contract():
```

Comment:
notice, we have to load the contract into `sys.modules` to make the registry work. (The entry point is the import path `packages.author.contracts...`)